### PR TITLE
Tutorial updates and feature flag

### PR DESCRIPTION
**Pull request by reichlfl**
This contains small fixes and updates for the tutorial framework.

For future development, I added a very simple (and non-final!) "How to take a capture"
Tutorial behind a feature flag `-enable_tutorials_feature`. Starting Orbit with this flag will
add a "Tutorials" menu entry to the main menu and provide a central place in code
to initialize and test future tutorials.

In addition, this changes the layout of the Tutorials and adjust them to work with our
current and planned Orbit UI:

- Tutorials use a cleaner look instead of the comic-like look
- Replaced the textual progress ("Tutorial A, Step 1/3") with radio-buttons to indicate
progress
- Controls for the tutorial are now at the bottom, including "close"

The close button is moved for the following reason: We need to add a dynamic content
margin to the main window to assure the tutorial controls do not overlap any UI elements
the user needs to click on, and having all those elements in a single row minimizes the 
space we loose by this.

Final layout still will need some discussion with UXR/D, so this is probably not the final
version.

3 commits:
- Small fixes to the basic framework
- Integration of sample tutorial into Orbit
- Layout changes and radio buttons to make the bottom margin work

image

### CODE REVIEW

File: OrbitQt/TutorialOverlay.cpp [hunk 1]
```diff
@@ -86,12 +85,43 @@ QRect TutorialOverlay::AbsoluteGeometry(QWidget* widget) const {
...
+  result.setBottomRight(result.bottomRight() + parent_geo.topLeft());
 
   return result;
 }
 
-TutorialOverlay::~TutorialOverlay() {}
+void TutorialOverlay::DeleteProgressIndicators() {
+  auto layout = ui_->progressLayout;
+  int count = layout->count() - 1;
+  for (int i = 1; i < count; ++i) {
```
Comment (pierricgimmig) [inline OrbitQt/TutorialOverlay.cpp:97]: Is there an off by one here? i starts at 1 and count is "layout->count() - 1" . Is that wanted?

**Review by pierricgimmig — APPROVED** [review]
Hi Florian, sorry for the delay, this somehow flew under my radar. The code looks really good to me, I did have one question however, please see comment. I'll try it out and let you know how it goes.

**Comment by pierricgimmig** [comment]
Oh actually I get a seg fault on Linux:

libQt5Core.so.5!QMetaObject::Connection::~Connection() (Unknown Source:0)
CreateTakeACaptureStepSetup(OrbitMainWindow*)::$_0::operator()(TutorialOverlay*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const(const class {...} * this) (/usr/local/google/home/pierric/git/orbit/OrbitQt/TutorialContent.cpp:21)
std::__invoke_impl<void, CreateTakeACaptureStepSetup(OrbitMainWindow*)::$_0&, TutorialOverlay*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::__invoke_other, CreateTakeACaptureStepSetup(OrbitMainWindow*)::$_0&, TutorialOverlay*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)(class {...} & __f,  __args) (/usr/include/c++/10/bits/invoke.h:60)
std::__invoke_r<void, CreateTakeACaptureStepSetup(OrbitMainWindow*)::$_0&, TutorialOverlay*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(CreateTakeACaptureStepSetup(OrbitMainWindow*)::$_0&, TutorialOverlay*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)(class {...} & __fn,  __args) (/usr/include/c++/10/bits/invoke.h:110)
std::_Function_handler<void (TutorialOverlay*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&), CreateTakeACaptureStepSetup(OrbitMainWindow*)::$_0>::_M_invoke(std::_Any_data const&, TutorialOverlay*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)(const std::_Any_data & __functor,  __args,  __args) (/usr/include/c++/10/bits/std_function.h:291)
std::function<void (TutorialOverlay*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>::operator()(TutorialOverlay*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const(const std::function<void (TutorialOverlay *, const std::__cxx11::basic_string<char> &)> * this, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > & __args, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > & __args) (/usr/include/c++/10/bits/std_function.h:622)
TutorialOverlay::StartActiveStep(TutorialOverlay * this) (/usr/local/google/home/pierric/git/orbit/OrbitQt/TutorialOverlay.cpp:131)
TutorialOverlay::StartSection(TutorialOverlay * this,  section) (/usr/local/google/home/pierric/git/orbit/OrbitQt/TutorialOverlay.cpp:168)
SetupDynamicInstrumentationTutorial(QMenu*)::$_2::operator()() const() (/usr/local/google/home/pierric/git/orbit/OrbitQt/TutorialContent.cpp:50)
QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, SetupDynamicInstrumentationTutorial(QMenu*)::$_2>::call(SetupDynamicInstrumentationTutorial(QMenu*)::$_2&, void**)() (/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs_impl.h:146)
QtPrivate::Functor<SetupDynamicInstrumentationTutorial(QMenu*)::$_2, 0>::call<QtPrivate::List<>, void>(SetupDynamicInstrumentationTutorial(QMenu*)::$_2&, void*, void**)() (/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs_impl.h:256)
QtPrivate::QFunctorSlotObject<SetupDynamicInstrumentationTutorial(QMenu*)::$_2, 0, QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)(int which, QtPrivate::QSlotObjectBase * this_, QObject * r, void ** a, bool * ret) (/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs_impl.h:443)
libQt5Core.so.5![Unknown/Just-In-Time compiled code] (Unknown Source:0)
libQt5Widgets.so.5!QAction::triggered(bool) (Unknown Source:0)
libQt5Widgets.so.5!QAction::activate(QAction::ActionEvent) (Unknown Source:0)
libQt5Widgets.so.5![Unknown/Just-In-Time compiled code] (Unknown Source:0)
libQt5Widgets.so.5!QMenu::mouseReleaseEvent(QMouseEvent*) (Unknown Source:0)
libQt5Widgets.so.5!QWidget::event(QEvent*) (Unknown Source:0)
libQt5Widgets.so.5!QMenu::event(QEvent*) (Unknown Source:0)
libQt5Widgets.so.5!QApplicationPrivate::notify_helper(QObject*, QEvent*) (Unknown Source:0)

File: OrbitQt/TutorialOverlay.cpp (hunk 1 above)
Comment (reichlfl) [reply OrbitQt/TutorialOverlay.cpp:97]: This is correct, the layout contains two horizontal spacers and the radio buttons are added in between. I should probably add a comment to explain that as it looks very suspicious.

**Comment by reichlfl** [comment]
> Oh actually I get a seg fault on Linux:

I've realized the same on Windows today, sorry for that! Apparently I did a small change in my last commit that I didn't properly test, and `QMetaObject::Connection` is either bugged or just not behaving as I expected. I removed that part of the code, it was only used in the "Test Tutorial" anyways.

**Pull request merged**